crates/transaction-pool/src/traits.rs

```diff
@@ -1111,12 +1112,17 @@ impl EthPoolTransaction for EthPooledTransaction {
     }
 }
 
-impl FromRecoveredTransaction for EthPooledTransaction {
-    fn from_recovered_transaction(tx: TransactionSignedEcRecovered) -> Self {
+impl TryFromRecoveredTransaction for EthPooledTransaction {
+    type Error = TryFromRecoveredTransactionError;
+
+    fn try_from_recovered_transaction(
+        tx: TransactionSignedEcRecovered,
+    ) -> Result<Self, Self::Error> {
         // CAUTION: this should not be done for EIP-4844 transactions, as the blob sidecar is
         // missing.
         let encoded_length = tx.length_without_header();
-        EthPooledTransaction::new(tx, encoded_length)
+        let transaction = EthPooledTransaction::new(tx, encoded_length);
```

Comment:
here I want to check if the type is OP deposit, if so return an error

Reply:
I'm not familiar enough with what an OP deposit means in this context. I added error handling for this, but unsure what specific error to return.

Reply:
we need another error variant for unsupported txtype, see comment above, then we need to check if the txtype is <=3 here, else return an error with that viarant